src/BenchmarksServer/Startup.cs

```diff
@@ -171,20 +172,34 @@ public static int Main(string[] args)
 
         private static string CloneAndRestore(string path, ServerJob job)
         {
-            var benchmarksDir = Git.Clone(path, _benchmarksRepoUrl);
+            // It's possible that the user specified a custom branch/commit for the benchmarks repo,
+            // so we need to add that to the set of sources to restore if it's not already there.
+            //
+            // Note that this is also going to de-dupe the repos if the same one was specified twice at
+            // the command-line (last first to support overrides).
+            var repos = new HashSet<Source>(job.Sources.Reverse(), SourceRepoComparer.Instance);
+            repos.Add(new Source() { Repository = _benchmarksRepoUrl });
 
             // Clone
+            string benchmarksDir = null;
             var dirs = new List<string>();
             foreach (var source in job.Sources)
             {
                 var dir = Git.Clone(path, source.Repository);
+                if (string.Equals(source.Repository, _benchmarksRepoUrl, StringComparison.OrdinalIgnoreCase))
```

Comment:
I think we should use a weaker check for repository equality, to support (for example) specifying a fork of benchmarks like `http://github.com/foo/benchmarks.git`.  We could use the text between the last slash character and the next dot character (if it exists).  Because I think the trailing `.git` is optional, and we'd want `aspnet/benchmarks.git` to match `aspnet/benchmarks`.